### PR TITLE
fix: limit port stream message size (object-sizeOf)

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Attempt at using analysis of the actual message object to determine if the request is too large or not. Uses the [object-sizeOf](https://github.com/miktam/sizeof) package as a base. Seems to be about 3x faster than the JSON.stringify approach

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26208?quickstart=1)

## **Related issues**

See: https://github.com/MetaMask/metamask-extension/pull/26204

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: shared/modules/sizeof.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          });
+        } else {
+          sizeOfObject(state, object);
+        }
+        return
+      default:
+        return
+    }
+  };
+}
```
Comment (jiexi) [inline shared/modules/sizeof.js:96]: we could also make this function not make the wallet unresponsive while it determines if a message is too large or not by yielding execution at various points during calculation to allow other requests to come in and be processed

**Comment by jiexi** [comment]
Could be also be changed to just count the number of references as opposed to the size of the references

File: shared/modules/sizeof.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  Int16Array: 2,
+  Uint16Array: 2,
+  Int32Array: 4,
+  Uint32Array: 4,
+  Float32Array: 4,
+  Float64Array: 8,
+};
+
+const isNodePlatform =
+  typeof process === 'object' && typeof require === 'function';
```
Comment (adonesky1) [inline shared/modules/sizeof.js:22]: 🤯 I didn't know this was how to check for this...

File: shared/modules/sizeof.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    if (Buffer.isBuffer(object)) {
+      state.size += object.length;
+    }
+
+    const objectType = typeof object;
+    switch (objectType) {
+      case 'string':
+        // https://stackoverflow.com/questions/68789144/how-much-memory-do-v8-take-to-store-a-string/68791382#68791382
+        state.size += isNodePlatform
```
Comment (adonesky1) [inline shared/modules/sizeof.js:65]: not understanding this? part are we creating a string here?

Comment (jiexi) [reply shared/modules/sizeof.js:65]: no, calculating the byte size in memory of the string based on it's length and what execution environment the script is running in

**Pull request closed without merging**